app/services/messages/expirer.rb

```diff
@@ -0,0 +1,28 @@
+module Messages
+  class Expirer
+    attr_reader :message
+
+    def initialize(message)
+      @message = message
+    end
+
+    def call
+      expire_message!
+      notify_to_external_interface!
+    end
+
+    private
+
+    def expire_message!
+      message.update!(
+        expired: true,
+        expired_at: Time.zone.now
+      )
+    end
+
+    def notify_to_external_interface!
+      interface = message.interface
+      Notifications::DiscordNotifier.new(message).notify_message_expiration! if interface.source == 'discord_guild'
+    end
```

Comment:
This is a great scenario for another pub/sub implementation. basically for these reasons:
- We need to avoid having discord-specific dependencies on the `app/services/` layer as much as possible. Anything that is discord-related should live under `app/integrations/discord/`. Only that way we ensure that the integration doesn't penetrate too deep into what privy is.
- What if in the future we want to integrate another provider, like slack or signal or telegram or whatever? We would need to add here one more dependency to this class for each provider we use. That's not ideal.

Let's broadcast an event here called `privy_message_expired` and add a listener in `app/integrations/discord/` to listen to the event and do whatever it needs to do on discord please.